### PR TITLE
Enhancing The Update Framework (Python - TUF) README

### LINKED ISSUE #2654: Enhancing The Update Framework (Python - TUF) README
**Description of issue or feature request**:
I would like to request an enhancement for the README of The Update Framework (TUF) repository. The goal is to improve the structure, clarity, and accessibility of the README to make it more informative and user-friendly.

**Current behavior**:
The current README lacks a clear structure, making it difficult for users to navigate and find relevant information quickly. Additionally, some sections could be more concise.

**Expected behavior**:
I propose the following changes to enhance the README:

1. **Add a Table of Contents**: Improve navigation and accessibility.
2. **Improve the Introduction**: Provide a clear and concise overview of TUF in 2 - 3 lines.
3. **Expand the "About TUF" Section**: Add more context about its significance and applications.
5. **Highlight Key Features**: Create a section listing key features and benefits.
6. **Reorganize the Documentation Section**: Enhance user-friendliness and navigation.
7. **Add Visuals**: Incorporate diagrams or images for better understanding.
8. **Change Logo Color**: Modify the logo from black to white to enhance visibility.
9. **Add Badges**: Include additional badges to highlight key aspects like version, license, tech-stack and community support.

### PR DESCRIPTION
**Pull request by shubhusion**
**Description of the changes being introduced by the pull request**:

1. **Table of Contents**: Added a Table of Contents to improve navigation and accessibility within the README.
   
2. **Logo Color**: Changed the logo color from black to white to enhance visibility and improve the overall aesthetic of the document.
   
3. **Additional Badges**: Included additional badges to highlight key aspects such as version, license, tech-stack, and community support, providing users with quick access to essential information.
   
4. **Back to Top Button**: Added a "Back to Top" button at the bottom of the README to facilitate easy navigation within the document.
   
5. **Installation Section**: Added an Installation section to provide users with clear guidance on how to install the software.

6. **Contact Section Enhancement**: Separated the Mailing List and CNCF Slack links for better readability and organization within the Contact section.

7. **CNCF Logo Addition**: Added the CNCF logo at the bottom of the README to acknowledge the project's affiliation with the Cloud Native Computing Foundation.

**Before:**
image

**After:**
image

Fixes #2654

Reviewers: 

@trishankatdatadog @JustinCappos

### CODE REVIEW

**Comment by shubhusion** [comment]
@trishankatdatadog @JustinCappos please review this PR for the issue #2654

**Comment by jku** [comment]
> 2\. **Logo Color**: Changed the logo color from black to white to enhance visibility and improve the overall aesthetic of the document.

This change makes the logo invisible with default GitHub settings (You are using dark theme on GitHub so see the opposite effect).

**Comment by shubhusion** [comment]
> > 2. **Logo Color**: Changed the logo color from black to white to enhance visibility and improve the overall aesthetic of the document.
> 
> This change makes the logo invisible with default GitHub settings (You are using dark theme on GitHub so see the opposite effect).

To ensure the logo is visible in both light and dark themes, a solution that automatically switches the logo based on the user's selected theme preference can be implemented. This strategy involves utilizing two different logo versions (one for light theme and one for a dark theme) and utilizing HTML and CSS to manage the theme-based switching.

**Comment by shubhusion** [comment]
@jku it works for both light and dark theme now. please have a look

File: README.md [hunk 1]
```diff
@@ -1,113 +1,143 @@
-# <img src="https://cdn.rawgit.com/theupdateframework/artwork/3a649fa6/tuf-logo.svg" height="100" valign="middle" alt="TUF"/> A Framework for Securing Software Update Systems
+<picture>
+  <source media="(prefers-color-scheme: dark)" srcset="https://github.com/shubhusion/python-tuf/blob/develop/tuf-horizontal-white.png">
+  <source media="(prefers-color-scheme: light)" srcset="https://github.com/shubhusion/python-tuf/blob/develop/tuf-horizontal-black.png">
+  <img alt="Shows an illustrated sun in light mode and a moon with stars in dark mode." src="https://github.com/shubhusion/python-tuf/blob/develop/tuf-horizontal-black.png">
+</picture>
```
Comment (jku) [inline README.md]: * alt text is incorrect
* I don't think you need to specify `source` for light theme: the default `img` is enough
* I think linking to pngs in cncf/artwork project would be totally fine, even preferable: if you do want to upload the pngs to this project then
   * please don't put them directly in the root directory
   * `https://github.com/shubhusion/python-tuf/` is not ok as url

File: README.md [hunk 2]
```diff
@@ -1,113 +1,143 @@
...
+# <div align="center">A Framework for Securing Software Update Systems</div>
 
 ![Build](https://github.com/theupdateframework/python-tuf/actions/workflows/ci.yml/badge.svg)
 [![Coveralls](https://coveralls.io/repos/theupdateframework/python-tuf/badge.svg?branch=develop)](https://coveralls.io/r/theupdateframework/python-tuf?branch=develop)
 [![Docs](https://readthedocs.org/projects/theupdateframework/badge/)](https://theupdateframework.readthedocs.io/)
 [![CII](https://bestpractices.coreinfrastructure.org/projects/1351/badge)](https://bestpractices.coreinfrastructure.org/projects/1351)
 [![PyPI](https://img.shields.io/pypi/v/tuf)](https://pypi.org/project/tuf/)
 [![OpenSSF Scorecard](https://api.securityscorecards.dev/projects/github.com/theupdateframework/python-tuf/badge)](https://api.securityscorecards.dev/projects/github.com/theupdateframework/python-tuf)
+[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
+[![License: Apache 2.0](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
```
Comment (jku) [inline README.md]: These links are already provided by github as clickable tabs just a few cm above this.

File: README.md [hunk 3]
```diff
@@ -1,113 +1,143 @@
...
+
+- [Introduction](#introduction)
+- [About The Update Framework](#about-the-update-framework)
+- [Key Features](#key-features)
+- [Documentation](#documentation)
+- [Installation](#installation)
+- [Contact](#contact)
+- [Security Issues and Bugs](#security-issues-and-bugs)
+- [License](#license)
+- [Acknowledgements](#acknowledgements)
```
Comment (jku) [inline README.md]: GitHub already provides an automatic  TOC in the tab bar now that you've added headers. I have to wonder if that would be enough (considering that one requires 0 maintenance)

File: README.md [hunk 4]
```diff
@@ -1,113 +1,143 @@
...
 
-----------------------------
-[The Update Framework (TUF)](https://theupdateframework.io/) is a framework for
-secure content delivery and updates. It protects against various types of
-supply chain attacks and provides resilience to compromise. This repository is a
-**reference implementation** written in Python. It is intended to conform to
-version 1.0 of the [TUF
-specification](https://theupdateframework.github.io/specification/latest/).
+---
```
Comment (jku) [inline README.md]: Now that you are using headers (good idea), I think these horizontal lines are just clutter: I think most of them can just be removed

File: README.md [hunk 5]
```diff
@@ -1,113 +1,143 @@
...
+      <img src="https://github.com/shubhusion/python-tuf/blob/develop/cncf-black.png" height="150" alt="TUF" style="background-color:white"/>
+    </picture>
+  </a>
+</div>
+
+
+---
+
+[![](https://img.shields.io/badge/Move%20to%20Top-%E2%86%91%20Back%20to%20Top-blue)](#a-framework-for-securing-software-update-systems)
+
```
Comment (jku) [inline README.md]: This duplicates functionality people already have in their browsers, except now we have to maintain that functionality. Let's not add things like this.

**Review by jku — CHANGES_REQUESTED** [review]
I've not looked at the text content changes(would appreciate someone else having a look). 

Structural changes look nice (like use of actual headers) but I left comments on the urls etc

**Comment by trishankatdatadog** [comment]
> @trishankatdatadog @JustinCappos please review this PR for the issue #2654

Sorry, I'm running out of B/W... please try someone else? @jku has been massively helpful here, thanks!

**Comment by lukpueh** [comment]
@jku already left a thorough review here. From what I see, the key improvement of this PR is to use the correct markup for the section headers. @shubhusion, do you plan to address the review comments?

**Comment by shubhusion** [comment]
> @jku already left a thorough review here. From what I see, the key improvement of this PR is to use the correct markup for the section headers. @shubhusion, do you plan to address the review comments?

@lukpueh yeah I will make changes as specified

File: README.md [hunk 6]
```diff
@@ -7,107 +12,91 @@
...
+<div align="center">
+  <a href="https://www.cncf.io/">
+    <picture>
+      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/shubhusion/python-tuf/blob/develop/cncf-white.png">
+      <source media="(prefers-color-scheme: light)" srcset="https://github.com/shubhusion/python-tuf/blob/develop/cncf-black.png">
+      <img src="https://github.com/shubhusion/python-tuf/blob/develop/cncf-black.png" height="150" alt="TUF" style="background-color:white"/>
+    </picture>
+  </a>
+</div>
```
Comment (jku) [inline README.md:102]: * links to "https://github.com/shubhusion/ are still not ok
* I also wonder if CNCF really needs a second level title and a huge logo in addition to the acknowledgement it already had. We could just tweak the previous CNCF reference in "Acknowledgements" to include the mention of being a graduated project?

**Review by jku — CHANGES_REQUESTED** [review]
* There are now unused pngs included in the PR: please remove those
* There are some issues with the CNCF section at the bottom, left a comment there

**Note to PR merger**: If the pngs are in any commit in the PR, this should be squash merged to prevent them from being part of git history

**Comment by jku** [comment]
I'm closing this since no action in 2 months. 

Please reopen if you get to it at a later date

**Pull request closed without merging**